lib/galaxy/model/__init__.py

```diff
@@ -3314,6 +3314,8 @@ def copy(self, parent_id=None, copy_tags=None, flush=True, copy_hid=True, new_na
 
     def copy_tags_to(self, copy_tags=None):
         if copy_tags is not None:
+            if isinstance(copy_tags, dict):
```

Comment:
I think I have a slight preference for typing copy_tags to be an Iterable of tags and change https://github.com/mvdbeek/galaxy/blob/4cde148b070a827dad4433601e9b814efc3a5255/lib/galaxy/tools/actions/__init__.py#L264-L278 to a list

Reply:
I am going to create an issue so we can target that in dev at some point. I hope that is okay. I agree this is an ugly hack - given the last change (#10761) broke things in subtle ways and didn't have test coverage for the underlying issue/fix I'm hesitant to change it again.